clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -146,30 +163,19 @@ public void close() throws IOException {
           for (BufferedBlockOutStream bos : mPreviousBlockOutStreams) {
             bos.close();
           }
-          canComplete = true;
+          return true;
         }
       } catch (IOException ioe) {
         handleCacheWriteException(ioe);
       }
     }
-
-    if (canComplete) {
-      FileSystemMasterClient masterClient = mContext.acquireMasterClient();
-      try {
-        masterClient.completeFile(mFileId);
-      } catch (TException e) {
-        throw new IOException(e);
-      } finally {
-        mContext.releaseMasterClient(masterClient);
-      }
-    }
-    mClosed = true;
+    return false;
```

Comment:
sure we should keep the same logic. I guess you can return `false` inside `(mCancled)` branch which makes the code easier to read.